src/tasks/deployment.ts

```diff
@@ -0,0 +1,116 @@
+import { promises as fs } from "fs";
+
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import {
+  parseCsvFile,
+  computeProofs,
+  removeSplitClaimFiles,
+  splitClaimsAndSaveToFolder,
+  DeploymentProposalSettings,
+  generateProposal,
+} from "../ts";
+
+import { defaultTokens } from "./ts/constants";
+import { defaultSafeDeploymentAddresses } from "./ts/safe";
+
+const OUTPUT_FOLDER = "./output/deployment";
+
+interface Args {
+  claims: string;
+  settings: string;
+}
+
+interface VirtualTokenSettings {
+  gnoPrice: string;
+  nativeTokenPrice: string;
+}
+
+export interface Settings
+  extends Omit<DeploymentProposalSettings, "virtualCowToken"> {
+  virtualCowToken: VirtualTokenSettings;
+}
+
+const setupDeployment: () => void = () => {
+  task(
+    "deployment",
+    `This script takes a list of user claims and deployment settings and produces:
+(1) the transactions that need to be executed from a Gnosis Safe to deploy the Cow DAO and the token contracts onchain, and
+(2) a list of all claims with corresponding proof in a format that is easy to handle by the frontend.`,
+  )
+    .addParam(
+      "claims",
+      "Path to the CSV file that contains the list of claims to generate.",
+    )
+    .addParam(
+      "settings",
+      "Path to the JSON file that contains the deployment settings.",
+    )
+    .setAction(generateDeployment);
+};
+
+export { setupDeployment };
+
+async function generateDeployment(
+  { claims: claimCsv, settings: settingsJson }: Args,
+  hre: HardhatRuntimeEnvironment,
+): Promise<void> {
+  const chainId = (await hre.ethers.provider.getNetwork()).chainId.toString();
+  if (chainId !== "1") {
+    throw new Error(
+      `This script must be run on mainnet. Found chainid ${chainId}`,
+    );
+  }
+
+  console.log("Processing input files...");
+  // TODO: validate settings
+  const inputSettings: Settings = JSON.parse(
+    await fs.readFile(settingsJson, "utf8"),
+  );
+  const claims = await parseCsvFile(claimCsv);
+
+  console.log("Generating Merkle proofs...");
+  const { merkleRoot, claims: claimsWithProof } = computeProofs(claims);
+
+  const settings = {
+    ...inputSettings,
+    virtualCowToken: {
+      ...inputSettings.virtualCowToken,
+      merkleRoot,
+      usdcToken: defaultTokens.usdc[chainId],
+      gnoToken: defaultTokens.gno[chainId],
+      wrappedNativeToken: defaultTokens.weth[chainId],
+    },
+  };
+  const { steps, addresses } = await generateProposal(
+    settings,
+    defaultSafeDeploymentAddresses(chainId),
+    hre.ethers,
+  );
+
+  console.log("Clearing old files...");
+  await fs.rm(`${OUTPUT_FOLDER}/addresses.json`, {
+    recursive: true,
+    force: true,
+  });
+  await fs.rm(`${OUTPUT_FOLDER}/steps.json`, { recursive: true, force: true });
+  await fs.rm(`${OUTPUT_FOLDER}/claims.json`, { recursive: true, force: true });
+  await removeSplitClaimFiles(OUTPUT_FOLDER);
```

Comment:
nit: tiny bit code duplication with test-deployment. Maybe we put these functions into a separate function. I need it my scripts as well..

Reply:
The only duplicated lines are:
```
  await fs.rm(`${OUTPUT_FOLDER}/claims.json`, { recursive: true, force: true });
  await removeSplitClaimFiles(OUTPUT_FOLDER);
```

The other two lines are new and there are different lines in the other script (because of `params.json`).
So I don't see a way to generalize it without making a generic function that takes files to be deleted.

Reply:
What is the other script for by the way? I would expect to reuse this when we introduce the xDAI deployment.